test/Splunk.OpenTelemetry.AutoInstrumentation.Tests/ServiceNameWarningTests.cs

```diff
@@ -0,0 +1,97 @@
+// <copyright file="ServiceNameWarningTests.cs" company="Splunk Inc.">
+// Copyright Splunk Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using Splunk.OpenTelemetry.AutoInstrumentation.Logging;
+
+namespace Splunk.OpenTelemetry.AutoInstrumentation.Tests;
+
+public class ServiceNameWarningTests : IDisposable
+{
+    private const string OtelServiceNameKey = "OTEL_SERVICE_NAME";
+    private const string OtelResourceAttributesKey = "OTEL_RESOURCE_ATTRIBUTES";
+
+    private Mock<ILogger> _logger = new Mock<ILogger>();
```

Comment:
Maybe instead of mocking you could create a wrapper/decorator/child to address this comment https://github.com/signalfx/splunk-otel-dotnet/pull/97#discussion_r1039308254 ?